include/libdnf5/conf/vars.hpp

```diff
@@ -70,6 +71,14 @@ struct Vars {
     /// @param prio Source/Priority of the value
     void set(const std::string & name, const std::string & value, Priority prio = Priority::RUNTIME);
 
+    /// @brief Set a variable to a value, only obtaining the value if needed using `get_value`
+    ///
+    /// @param name Name of the variable
+    /// @param get_value Function that returns the (optional) value for the variable
+    /// @param prio Source/Priority of the value
+    void set_lazy(
```

Comment:
I am not sure whether it is a good idea to detect vars not at the same time point. Right not it happen at Base::setup() which is a mandatory step. Is there a particular reason to do it this way? Originally we were thinking about locking vars like with configuration, but the feature has low priority therefore it is not yet implemented.

Reply:
`set_lazy` doesn't actually defer setting the var, it's the same as `set` but only evaluates the value to be set if it's necessary to do so, i.e. when the incoming priority is greater than the existing priority.

It's a somewhat common pattern, often you will see a `log.Println(() => string)` function that only evaluates the lambda if necessary.

I added it to deduplicate some of the code in `detect_vars`.

Reply:
Thank you very much for explanation. Do you think we need it as API?

Reply:
You could just move it to the `private` section of the class if it's only used in the Vars class.

Reply:
OK, done.